data/Record.js

```diff
@@ -20,29 +22,57 @@ export class Record {
     id;
     /** @member {(string|number)} */
     parentId;
-    /** @member {Store} */
-    store;
+    /** @member {Record} */
+    originalRecord;
     /** @member {Object} */
     raw;
+    /** @member {Object} */
+    data;
+    /** @member {Store} */
+    store;
     /** @member {boolean} - flag set post-construction by Store on summary recs - for Hoist impl. */
     xhIsSummary;
+    /** @member {String[]} - unique path within hierarchy - for ag-Grid impl. */
+    xhTreePath;
+
+    /** @returns {boolean} */
+    get isNew() {
+        return this.originalRecord === null;
+    }
 
-    _xhTreePath = null;
+    /** @returns {boolean} */
+    get isDirty() {
+        return this.originalRecord !== this;
+    }
+
+    /** @returns {boolean} */
+    get isModified() {
+        return !this.isNew && this.isDirty;
+    }
+
+    /** @returns {Object|null} */
```

Comment:
perhaps make this a method, since it is doing a fair bit of work and allocation?